### DISCUSSION TITLE
Restart from Last step

### DISCUSSION THREAD

**Discussion by wangzhaohao** [idaholab/moose#27405] [Q&A General] [answered]
Hi, I am using `restart` in MOOSE
I want to use the previous solid mechanics output, which is .e file.
I have set the `disp_x disp_y disp_z`, there seems no problem.
But I meet a problem is the stress is zero, what should i do to transfer it?
Thanks for you help!
![image](https://github.com/idaholab/moose/assets/51054734/c4e11fd7-c520-45f9-9e8c-cbb1deaa199f)

**Comment by GiudGiud** [comment]
Hello

You cannot use restart from exodus for solid mechanics. You have to use the checkpoint system for restarting

Guillaume

**Reply by wangzhaohao** [reply]
thanks, i find `checkpoint` in this [link](https://mooseframework.inl.gov/application_usage/restart_recover.html)
and a question is there no mention of how to use. Can you give me some examples or other links?

**Reply by GiudGiud** [reply]
You add a Checkpoint object to your Outputs block to create the checkpoint
and use the `restart_file_base` parameter in the Problem to use it.

there will be examples in the repository. `grep -r restart_file_base` will find them

**Reply by wangzhaohao** [reply]
Thanks, I find some examples, and I do as follow:
```Assembly
[Problem]
   restart_file_base = XAPR_heatsource_out_fuelRod0_cp/LATEST
   force_restart = true
[]
```
the `XAPR_heatsource_out_fuleRod0_cp` is a subApp, So I use `force_restart`
But seems doesn't work
```Shell
WARNING! There are options you set that were not used!
WARNING! could be spelling mistake, etc!
There is one unused database option. It is:
Option left: name:-i value: pulse_project/pulse_heat.i source: command line
```
I use the command in zsh is ` mpiexec -n 40 ./pulse-opt -i pulse_project/pulse_heat.i`

**Reply by wangzhaohao** [reply, marked as answer]
Oh, I find the reason
**the end time should follow the LATEST**
